cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,243 @@
+
+* The [Attributes](#2-attributes) section describes the representation and
+data type mappings for CloudEvents context attributes.
+
+* The [Data](#3-data) section defines the container for the data portion of a
+CloudEvent.
+
+* The [Envelope](#4-envelope) section defines an XML container for CloudEvents
+attributes and an associated media type.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+may be packaged into a single XML document.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This specification is deliberately verbose in favor of readability and
+deterministic processing.
+
+Preservation of attribute value type information is supported allowing
+custom extensions to be communicated without value type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to the XML schema types as follows :
+
+| CloudEvents   |  XML Schema Type| Notes |
+| :-------------| :---------------------------------------------------------- |
+| Boolean       | [xs:boolean][xml-primitives] | |
+| Integer       | [xs:int][xml-primitives] | |
+| String        | [xs:string][xml-primitives] | |
+| Binary        | [xs:base64Binary][xml-primitives] | |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each context attribute is represented as an XML element whose name exactly matches that of
+a required or optional CloudEvent attribute, extension attribute names `MUST` adhere and
+align with the conventions in defined in the [cloud event specification][ce-spec].
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension attributes `SHOULD` be expressed with the appropriate `xsi:type` to allow them
+to be exchanged without loss of type information. Extension attributes with no `xsi:type`
+discriminator `SHOULD` be interpreted with an implied type of `xs:string`
+
+``` xml
+    ...
+    <id>AAABBBCCCNNN0000</id>
+    ..
+    <time>2021-08-14T-08:00</time>
+    ..
+    <myextension xsi:type="xs:string">my extension value</myextension>
+    <myboolean xsi:type="xs:boolean">false</myboolean>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent follows a similar model to that employed by
+the [JSON Format specification][json-format]. The elmenent names used are capitalized to avoid
+collision with CloudEvent attribute names.
+
+### 3.1 Binary Data
+
+Carried in an element with an defined type of `xs:base64Binary`
+
+``` xml
+<Data xsi:type="xs:base64Binary">.........</Data>
```

Comment:
should it be "data" instead of "Data" ?

Reply:
Uppercase to avoid collision with context attributes

Reply:
`data` is (sort of) a context attribute though. That's why the example [here](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#example) makes it look like "data" is just like all other attributes

Reply:
That change was recently committed to the specification.